File: Loenn/entities/king_ddd_dummy.lua
local kingDDDDummy = {}

kingDDDDummy.name = "MaggyHelper/KingDDDDummy"
kingDDDDummy.depth = 0
kingDDDDummy.texture = "characters/kirby/king_dedede/idle00"
kingDDDDummy.justification = {0.5, 1.0}

kingDDDDummy.fieldInformation = {
    facing = {
        fieldType = "integer",
        options = {-1, 1},
        editable = false
    },
    animation = {
        fieldType = "string",
        options = {
            "idle",
            "walk",
            "happy",
            "talk"
        },
        editable = true
    },
    scale = {
        fieldType = "number",
        minimumValue = 0.1,
        maximumValue = 3.0
    },
    alpha = {
        fieldType = "number",
        minimumValue = 0.0,
        maximumValue = 1.0
    },
    isVisible = {
        fieldType = "boolean"
    },
    playAnimationOnSpawn = {
        fieldType = "boolean"
    },
    autoFollow = {
        fieldType = "boolean"
    },
    followDelay = {
        fieldType = "number",
        minimumValue = 0.0,
        maximumValue = 2.0
    }
}

kingDDDDummy.placements = {
    {
        name = "normal",
        data = {
            facing = 1,
            animation = "idle",
            scale = 1.0,
            alpha = 1.0,
            isVisible = true,
            playAnimationOnSpawn = false,
            autoFollow = true,
            followDelay = 0.95
        }
    }
}

function kingDDDDummy.sprite(room, entity)
    local facing = entity.facing or 1
    local scale = entity.scale or 1.0
    local alpha = entity.alpha or 1.0
    
    return {
        texture = "characters/kirby/king_dedede/idle00",
        x = entity.x,
        y = entity.y,
        justificationX = 0.5,
        justificationY = 1.0,
        scaleX = facing * scale,
        scaleY = scale,
        color = {1.0, 1.0, 1.0, alpha}
    }
end

return kingDDDDummy

File: framework/extracted/Ahorn/entities/KingDDDDummy.jl
module KingDDDDummy

using ..Ahorn, Maple

@mapdef Entity "MaggyHelper/KingDDDDummy" KingDDDDummy(x::Integer, y::Integer, alpha::Number=1.0, animation::String="idle", autoFollow::Bool=true, facing::Integer=1, followDelay::Number=0.95, isVisible::Bool=true, playAnimationOnSpawn::Bool=false, scale::Number=1.0)

const placements = Ahorn.PlacementDict(
    "normal" => Ahorn.EntityPlacement(KingDDDDummy)
)

function Ahorn.selection(entity::KingDDDDummy)
    return Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::KingDDDDummy, room::Maple.Room)
    Ahorn.drawSprite(ctx, "characters/kirby/Maggy/DesoloZantas/king_dedede/idle00", entity.x, entity.y)
end

end
